pytket_qir/qir.py

```diff
@@ -400,11 +480,79 @@ def condition_zero_block():
                 one=lambda: condition_one_block(),
                 zero=lambda: condition_zero_block(),
             )
+        elif isinstance(op, WASMOp):
+            cregs = _retrieve_registers(circ.bits, BitRegister)
+            args = command.args
+
+            inputs: List[str] = []
+            outputs: List[str] = []
+            for reglist, sizes in [
+                (inputs, op.input_widths),
+                (outputs, op.output_widths),
+            ]:
+                for in_width in sizes:
+                    assert in_width >= 0
+                    wasm_bits = args[:in_width]
+                    args = args[in_width:]
+                    regname = wasm_bits[0].reg_name
+                    if wasm_bits != list(cregs[regname]):
+                        WASMError("WASM ops must act on entire registers.")
+                    reglist.append(regname)
+
+            try:
+                bit_reg = circ.get_c_register(inputs[0])
+            except RuntimeError:
+                bit_reg = BitRegister("wasm_input", 0)
```

Comment:
1. Are you sure you would get `RuntimeError` if the index is out of range, not `IndexError`?
2. It would be good to add a test covering this edge case.
3. Will this still work if the circuit contains two zero-input WASM boxes? Won't there then be a clash of register names?
4. If the answer to (3) is yes, or if no and we have to have some unique register name, can we just use that solution in all cases instead of special-casing the 0 case?

Reply:
1. Is not an out-of-range error but a non-existent name. At the very least, `inputs` will be empty.
2. Certainly. I can do that.
3. This register is only created for the purposes of initialising the ssa variable to 0 when there is no input to the WASMOp. I don't think it really matters if there are multiple zero-input WASM boxes. And it is not reused anywhere else.
4. I don't think we need a unique register here.

Reply:
You're right as always. Done.